### PR TITLE
Create a standard Builtins module

### PR DESCRIPTION
**Pull request by e-kayrakli**
This PR creates a new Builtins module and moves some procs there.

Resolves https://github.com/Cray/chapel-private/issues/628
Resolves https://github.com/Cray/chapel-private/issues/939

As of this PR `standard/Builtins.chpl` includes:

- All the content from `Assert.chpl`
- `compiler{Warning,Error,Assert}`
- `proc exit()`
  - This was the main motivation for this PR. This move closes a memory leak.
    https://github.com/Cray/chapel-private/issues/628

More details of changes:

- Add `standard/Builtins.chpl`
- Deprecate `standard/Assert.chpl`
- Add header documentation in `Builtins.chpl` and adjust chpldoc Makefile to
  include this new module.
- Remove stuff that are included in this module from `UtilMisc_ForDocs.chpl`
- Stop checking the location of `__primitive("warning")` to suppress internal
  warnings that are based on dynamic dispatch.

  Before this, `compilerWarning` was in an internal module and thus, so was
  `__primitive("warning")` the check in question used to test both of these
  calls' (the call to `compilerWarning` *and* the call to
  `__primitive("warning")` within. I believe that this check didn't need to
  check for the primitive's location, but we got away with it because it also
  happened to be in an internal module. With this PR, we move the call to the
  primitive to a standard module and start to see some warnings.

- Update the meta-doc for standard modules (removing the statement that says
  `IO` is included by default, which has not been true for a while now)

- Bunch of test updates

  For one of them I changed filenames and thought I used `git mv` but diff says
  I am wrong. Sorry about the confusion.

Test:
- [x] `make docs` runs cleanly
- [x] `release/examples` with standard
- [x] standard
- [x] gasnet

### CODE REVIEW

File: modules/standard/Assert.chpl [hunk 1]
```diff
@@ -2,69 +2,25 @@
  * Copyright 2020 Hewlett Packard Enterprise Development LP
  * Copyright 2004-2019 Cray Inc.
  * Other additional copyright holders may be indicated within.
- * 
+ *
```
Comment (bradcray) [inline modules/standard/Assert.chpl:5]: This makes me wonder whether our copyright header will choke on this (or should have choked on the previous version?)

File: modules/standard/Assert.chpl [hunk 2]
```diff
@@ -2,69 +2,25 @@
...
-  Support for simple assert() routines.
-
-  .. note:: All Chapel programs automatically ``use`` this module by default.
-            An explicit ``use`` statement is not necessary.
-
-  In the current implementation, these asserts never become no-ops.  That is,
-  using them will always incur execution-time checks.
-
-*/
 module Assert {
```
Comment (bradcray) [inline modules/standard/Assert.chpl:21]: I think that @ben-albrecht (?) has had some text to use for modules that are being deprecated so that if one starts leafing through different versions of a module in the online chpldocs, they will see a description of it disappearing rather than having it simply not be there in some release (?).

**Comment by bradcray** [comment]
I didn't really review this so much as give it a quick glance as a study break (i.e., someone else still should).

File: modules/standard/Assert.chpl (hunk 1 above)
Comment (e-kayrakli) [reply modules/standard/Assert.chpl:5]: ugh, I haven't realized this. Probably we have two "versions" w.r.t whitespace and I picked up the wrong one when I bring back the Assert module. I am not sure what is it that you are wondering exactly, but I'll make sure that there is no diff here for PR cleanliness

Comment (bradcray) [reply modules/standard/Assert.chpl:5]: There's a script that makes sure all key files have the right copyright header.  I'm not certain how sensitive it is to trailing whitespace or not.

Comment (e-kayrakli) [reply modules/standard/Assert.chpl:5]: Ah I see. I'll look that up

File: modules/standard/Assert.chpl (hunk 2 above)
Comment (ben-albrecht) [reply modules/standard/Assert.chpl:21]: https://github.com/chapel-lang/chapel/blob/master/doc/rst/developer/bestPractices/Deprecation.rst

```
    Foo module short description.

    Warning

    This module has been deprecated - please use :mod:`bar` instead.

    Foo module long description.
```

File: modules/standard/Builtins.chpl [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+  This module contains built-in functions.
+
+  .. note:: All Chapel programs automatically ``use`` this module by default.
```
Comment (ben-albrecht) [inline modules/standard/Builtins.chpl:24]: Does this render correctly? If so, TIL! I always thought the content of the note had to come after a newline.

File: test/modules/sungeun/init/printModuleInitOrder.good [hunk 4]
```diff
@@ -3,7 +3,6 @@ Initializing Modules:
     OwnedObject
     Atomics
      MemConsistency
-    NetworkAtomics
```
Comment (ben-albrecht) [inline test/modules/sungeun/init/printModuleInitOrder.good:6]: Why is this removed here?

File: modules/standard/Builtins.chpl (hunk 3 above)
Comment (e-kayrakli) [reply modules/standard/Builtins.chpl:24]: Apparently :)

Screen Shot 2020-05-07 at 4 48 36 PM

File: test/modules/sungeun/init/printModuleInitOrder.good (hunk 4 above)
Comment (e-kayrakli) [reply test/modules/sungeun/init/printModuleInitOrder.good]: Ah thanks for catching that, probably I copied the exec.out.tmp to wrong good file.
